cluster.c

```diff
@@ -597,18 +699,34 @@ RRStatus ShardingInfoValidateShardGroup(RedisRaftCtx *rr, ShardGroup *new_sg)
 
 RRStatus ShardingInfoUpdateShardGroup(RedisRaftCtx *rr, ShardGroup *new_sg)
 {
-    ShardingInfo *si = rr->sharding_info;
+    if (!strncmp(rr->snapshot_info.dbid, new_sg->id, sizeof(new_sg->id))) {
```

Comment:
memcmp is probably better, it's fixed length anyway.

Reply:
switched